extensions/toggl-track/src/components/CreateTimeEntryForm.tsx

```diff
@@ -7,33 +7,42 @@ import {
   showToast,
   ToastStyle,
   clearSearchBar,
+  getPreferenceValues,
 } from "@raycast/api";
 import toggl from "../toggl";
 import { storage } from "../storage";
 import { Project } from "../toggl/types";
 import { useAppContext } from "../context";
 import { useMemo, useState } from "react";
 
+interface Preferences {
+  defaultProject: string;
+}
+
+const preferences: Preferences = getPreferenceValues();
+
 function CreateTimeEntryForm({ project, description }: { project?: Project; description?: string }) {
   const navigation = useNavigation();
   const { projects, tags, isLoading, projectGroups } = useAppContext();
   const [selectedProject, setSelectedProject] = useState<Project | undefined>(project);
   const [selectedTags, setSelectedTags] = useState<string[]>([]);
 
-  async function handleSubmit(values: { description: string }) {
+  async function handleSubmit(values: { description: string; start: string; end: string; }) {
     try {
       await toggl.createTimeEntry({
         projectId: selectedProject?.id,
         description: values.description,
         tags: selectedTags,
+        start: values.start,
+        duration: Math.trunc((new Date(values.end).getTime() - new Date(values.start).getTime()) / 1000)
```

Comment:
Doesn't the Toggl Track API passing the end time directly as `end`? This is safer than calculating the duration. Pass in `end: values.end` instead?